libnetwork/network.go

```diff
@@ -1565,7 +1570,10 @@ func (n *network) requestPoolHelper(ipam ipamapi.Ipam, addressSpace, preferredPo
 		}
 
 		// Check for overlap and if none found, we have found the right pool.
-		if _, err := netutils.FindAvailableNetwork([]*net.IPNet{pool}); err == nil {
+		if _, err := netutils.FindAvailableNetwork([]*net.IPNet{pool}, func(nw *net.IPNet) bool {
+			_, maskSize := nw.Mask.Size()
+			return maskSize == expectedMaskSize
+		}); err == nil {
```

Comment:
This call to `netutils.FindAvailableNetwork` is being passed a list of length 1. It's used to check whether the pool—singular—returned from `ipam.RequestPool` would conflict with routes in use on the host. The call to `ipam.RequestPool` specifies the address family (IPv4 vs. IPv6), so the question is: why do we have to double-check that `ipam.RequestPool` respected the request? Seems like we should be fixing the root cause, the bug in the IPAM allocator which allocates IPv6 pools when an IPv4 pool is requested.